src/components/TodoItem.tsx

```diff
@@ -0,0 +1,185 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todo: Todo;
+  isTemp?: boolean;
+  onDelete: (todoId: number) => void;
+  isDeleting?: boolean;
+  onToggleStatus: (todoId: number, newStatus: boolean) => void;
+  isLoading?: boolean;
+  loading?: boolean;
```

Comment:
```suggestion
  isLoading?: boolean;
```
You can get rid of additional loading prop